src/System.Private.Reflection.Core/src/System/Reflection/Runtime/Assemblies/NativeFormat/NativeFormatRuntimeAssembly.cs

```diff
@@ -165,6 +166,23 @@ public sealed override int GetHashCode()
             return Scope.Handle.GetHashCode();
         }
 
+        public sealed override string Location
+        {
+            get
+            {
+                // While we can't return a path to a file with IL, this will satisfy the apps that use this api
+                // to get the exe's install directory or a name to display in help text.
+                string location = RuntimeAugments.TryGetFullPathToMainApplication();
```

Comment:
Cc @dotnet/corert-contrib if anyone has any good ideas for this.

This concerns me for two reasons:

* The `EndsWith`/replace hack is Windows specific.
* `TryGetFullPathToMainApplication` hasn't been updated ever since we enabled the shared library feature and will in fact give you the path to the shared library in our default configurations. We should really just delete it, but it's used as a fallback implementation for a piece of non-shipping resources logic.